src/method.c

```diff
@@ -275,9 +274,11 @@ JL_DLLEXPORT jl_code_info_t *jl_code_for_staged(jl_method_instance_t *linfo)
     jl_expr_t *ex = NULL;
     jl_value_t *linenum = NULL;
     jl_svec_t *sparam_vals = env;
-    jl_method_instance_t *generator = linfo->def.method->generator;
+    jl_value_t *generator = linfo->def.method->generator;
+    jl_method_t *gen_meth = jl_gf_mtable(generator)->defs.leaf->func.method;
```

Comment:
Seems like the names either need to be a characteristic of the original function being optimized, or the generator should be just a single Method object (rather than a function)